### PR TITLE
Add read/write error counts to Bigquery

### PR DESCRIPTION
**Pull request by jimmyfigiel**
This change exposes metrics for the number of read errors from and write errors to Bigquery for prometheus to scrape.

Partially fixes Issue #9

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -81,13 +81,28 @@ var (
...
+	writeErrors = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "storage_bigquery_write_error_count",
+			Help: "Total number of write errors to Bigquery.",
+		},
+	)
+	readErrors = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "storage_bigquery_read_error_count",
+			Help: "Total number of read errors to Bigquery.",
```
Comment (Smiley73) [inline main.go:93]: --> "from BigQuery"

File: main.go [hunk 2]
```diff
@@ -81,13 +81,28 @@ var (
 		},
 		[]string{"remote"},
 	)
+	writeErrors = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "storage_bigquery_write_error_count",
+			Help: "Total number of write errors to Bigquery.",
```
Comment (Smiley73) [inline main.go:87]: Bigquery --> BigQuery

File: main.go [hunk 3]
```diff
@@ -81,13 +81,28 @@ var (
...
 )
 
 func init() {
 	prometheus.MustRegister(receivedSamples)
 	prometheus.MustRegister(sentSamples)
 	prometheus.MustRegister(failedSamples)
 	prometheus.MustRegister(sentBatchDuration)
+	prometheus.MustRegister(writeErrors)
+	prometheus.MustRegister(readErrors)
+
```
Comment (Smiley73) [inline main.go:105]: Remove the extra newline

File: main.go [hunk 4]
```diff
@@ -278,6 +300,7 @@ func sendSamples(logger log.Logger, w writer, timeseries []*prompb.TimeSeries) {
 	if err != nil {
 		level.Warn(logger).Log("msg", "Error sending samples to remote storage", "err", err, "storage", w.Name(), "num_samples", len(timeseries))
 		failedSamples.WithLabelValues(w.Name()).Add(float64(len(timeseries)))
+		writeErrors.Inc()
 	}
 	sentSamples.WithLabelValues(w.Name()).Add(float64(len(timeseries)))
```
Comment (Smiley73) [inline main.go:305]: I know this is from previous code, but can you move this line into an `else` statement?
We shouldn't count the sample if it wasn't successful

File: main.go [hunk 5]
```diff
@@ -81,13 +81,28 @@ var (
 		},
 		[]string{"remote"},
 	)
+	writeErrors = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "storage_bigquery_write_error_count",
```
Comment (seanmalloy) [inline main.go:86]: The other counter metrics end in `_total`, but this one ends in `_count`. For consistency I'd name this metric `storage_bigquery_write_error_total`

File: main.go [hunk 6]
```diff
@@ -81,13 +81,28 @@ var (
...
+			Help: "Total number of write errors to Bigquery.",
+		},
+	)
+	readErrors = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "storage_bigquery_read_error_count",
```
Comment (seanmalloy) [inline main.go:92]: Same here. Consider naming this metric `storage_bigquery_read_error_total`

Comment (seanmalloy) [reply main.go]: I think errors should be plural. it should be `storage_bigquery_read_errors_total`. I missed this the first time I reviewed it.

File: main.go (hunk 5 above)
Comment (seanmalloy) [reply main.go]: Same here `errors` should be plural. I missed this in the first review.

**Pull request merged**